File: client-app/src/app/models/project.ts
export interface IProject {
    id: string;
    title: string;
    category: string;
    image?: string;
}

export class Project implements IProject {
    constructor(
        id: string, 
        title: string,
        category: string,
        image: string) {
        this.id = id;
        this.title = title;
        this.category = category;
        this.image = image;
    }
    id: string;
    title: string;
    category: string;
    image?: string;
}

export interface IProjectDetail extends IProject {
    content: string;
}

export class ProjectDetail extends Project implements IProjectDetail {
    constructor(
        id: string, 
        title: string,
        category: string,
        content: string) {
        super(id, title, category, '');
        this.content = content;
    }
    content: string;
}

export interface Photo {
    id: string;
    url: string;
}

File: PersonalWebsite/Domain/Project.cs
using System.ComponentModel.DataAnnotations;

namespace Domain
{
    public class Project
    {
        public Guid Id { get; set; }
        public string Title { get; set; }
        public string Description { get; set; }
        public string Content { get; set; }
        public string Image { get; set; }
        public string Category { get; set; }
        public ICollection<string> Tags { get; set; }
    }
}
